pandas/tests/io/formats/test_format.py

```diff
@@ -3241,12 +3241,12 @@ def test_repr_html_ipython_config(ip):
 
 @pytest.mark.parametrize("method", ["to_string", "to_html", "to_latex"])
 def test_filepath_or_buffer_arg(
-    float_frame, method, filepath_or_buffer, assert_filepath_or_buffer_equals
+    float_frame, method, filepath_or_buffer, assert_filepath_or_buffer_equals, encoding
```

Comment:
thanks for updating this. can you also parametrize encoding, probably None, "utf-8", "gbk" and "foo".

Reply:
and we should raise when encoding is specified and it's buf (or None) not a string path or pathlike (i.e. Path). xref https://github.com/pandas-dev/pandas/issues/28766#issuecomment-538413490

if you use add the `filepath_or_buffer_id` fixture to the parameter list, will make this simpler, see `assert_filepath_or_buffer_equals` fixture for example.

Reply:
> thanks for updating this. can you also parametrize encoding, probably None, "utf-8", "gbk" and "foo".
Is this "foo" encoding valid. I cannot find it.

Reply:
unlike the buf parameter, it is probably better to test for a bad argument to encoding in the same test as the valid buf parameters since a raising a message about specifiying the encoding parameter with buf or None should probably take precidence over raising a message about an invalid encoding.

the builtin open should raise that so no need to add to code, just check when the message is raise.

so I think the test would look something like

```
if filepath_or_buffer_id not in ["string", "pathlike"] and encoding is not None:
   with pytest.raises(ValueError, match = "<insert custom message here>"):
       do test
elif encoding=="foo"
   with pytest.raises(ValueError, match = "<message from builtin open command about invalid encoding >"):
     do test
else:
  do test
  assert_filepath_or_buffer_equals(expected)
```